notebooks/ntsb-duckdb.ipynb

```diff
@@ -0,0 +1,340 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pyarrow.fs\n",
+    "from ray.data import ActorPoolStrategy\n",
+    "import sycamore\n",
+    "import json\n",
+    "from sycamore.functions.tokenizer import HuggingFaceTokenizer\n",
+    "from sycamore.llms import OpenAIModels, OpenAI\n",
+    "from sycamore.transforms import COALESCE_WHITESPACE\n",
+    "from sycamore.transforms.merge_elements import MarkedMerger\n",
+    "from sycamore.transforms.partition import SycamorePartitioner\n",
+    "from sycamore.transforms.extract_schema import OpenAISchemaExtractor, OpenAIPropertyExtractor\n",
+    "from sycamore.transforms.embed import SentenceTransformerEmbedder"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sycamore.data.document import Document\n",
+    "from dateutil import parser\n",
+    "def convert_timestamp(doc: Document) -> Document:\n",
+    "    if \"dateAndTime\" not in doc.properties['entity'] and \"dateTime\" not in doc.properties['entity']:\n",
+    "        return doc\n",
+    "    raw_date: str = doc.properties['entity'].get('dateAndTime') or doc.properties['entity'].get('dateTime')\n",
+    "    raw_date = raw_date.replace(\"Local\", \"\")\n",
+    "    parsed_date = parser.parse(raw_date, fuzzy=True)\n",
+    "    extracted_date = parsed_date.date()\n",
+    "    doc.properties['entity']['day'] = extracted_date.day\n",
+    "    doc.properties['entity']['month'] = extracted_date.month\n",
+    "    doc.properties['entity']['year'] = extracted_date.year\n",
+    "    if parsed_date.utcoffset():\n",
+    "        doc.properties['entity']['dateTime'] = parsed_date.isoformat()\n",
+    "    else:\n",
+    "        doc.properties['entity']['dateTime'] = parsed_date.isoformat() + \"Z\"\n",
+    "    \n",
+    "    return doc"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "paths = [\"s3://aryn-public/ntsb/\"]\n",
+    "fsys = pyarrow.fs.S3FileSystem(region=\"us-east-1\", anonymous=True)\n",
+    "\n",
+    "llm = OpenAI(OpenAIModels.GPT_3_5_TURBO.value)\n",
+    "tokenizer = HuggingFaceTokenizer(\"thenlper/gte-small\")\n",
+    "\n",
+    "ctx = sycamore.init()\n",
+    "\n",
+    "ds = (\n",
+    "    ctx.read.binary(paths, binary_format=\"pdf\", filesystem=fsys)\n",
+    "    # Parition with the sycamore partitioner, pulling out tables and images. ActorPoolStrategy(size=3) works best on my particular hardware\n",
```

Comment:
I wouldn't use the word "my" here. Also, where is this ActorPoolStrategy field in the example? If it's not there as default, why are we saying it?

Reply:
Specifying it would help allow you to run queries at the maximum parallelism your machine can support. But I agree, this is additional not needed complexity. Have removed it now